src/App.js

```diff
@@ -26,40 +27,109 @@ const generateSquares = () => {
 };
 
 const App = () => {
-  // This starts state off as a 2D array of JS objects with
-  // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
-
+  const [ticDict, setTicDict] = useState({});
+  const [turn, setTurn] = useState(PLAYER_1);
+  const [winner, setWinner] = useState(null);
   // Wave 2
   // You will need to create a method to change the square
   //   When it is clicked on.
   //   Then pass it into the squares as a callback
 
+  const toggleTurn = () => {
+    let newTurn = '';
+    if (turn === PLAYER_1) {
+      newTurn = PLAYER_2;
+    } else {
+      newTurn = PLAYER_1;
+    }
+    setTurn(newTurn);
+  };
+
+  const onClickCallback = (squareID) => {
+    const newSquares = [...squares];
+    newSquares.forEach((row) => {
+      row.forEach((square) => {
+        if (square.id === squareID && square.value === '') {
+          square.value = turn;
+          ticDict[squareID] = turn;
```

Comment:
This variable isn't being used correctly as a state variable.  A state variable should be treated as if it is _const_ at all times, and changes should be made through the corresponding set state function.  To change a state var, make a copy, make the changes to the copy, and then call the set state function and pass in the changed copy.  This can cause some issues if you want to use the new version right away, but you can bypass this by using the copied version (ie, pass the copied version to checkForWinner).

Reply:
Hi Jasmine, thank you for the feedback. Can you clarify which variable isn't being used correctly as a state variable here? Do you mean that we shouldn't be writing directly to ticDict? We should be writing to  copy of ticDict and then using a set function to update it?